src/internal/observable/fromPromise.ts

```diff
@@ -1,3 +1,30 @@
-import {  PromiseObservable  } from './PromiseObservable';
+import { Observable } from '../Observable';
+import { IScheduler } from '../Scheduler';
+import { Subscription } from '../Subscription';
 
-export const fromPromise = PromiseObservable.create;
\ No newline at end of file
+export function fromPromise<T>(input: Promise<T>, scheduler: IScheduler) {
+  if (!scheduler) {
+    return new Observable<T>(subscriber => {
+      input.then(value => {
+        subscriber.next(value);
```

Comment:
Any reason why there is no longer the optimisation of becoming a scalar observable?